File: StrategyBot.Game.Logic/MainMenuController.fs
namespace StrategyBot.Game.Logic

open StrategyBot.Game.Core
open StrategyBot.Game.Core.Controllers
open StrategyBot.Game.Core.Controllers.Abstractions
open StrategyBot.Game.Core.Controllers.Handlers

[<Controller(isMainController = true)>]
type MainMenuController() =
    inherit ControllerBase()

    [<Action("screens.main_menu.say_my_name")>]
    member __.SayMyName(player: PlayerState) = __.Answer("Your name is " + player.SocialId)

    [<Action("screens.main_menu.attack")>]
    member __.AttackAction() = __.Answer "test"

    [<Default>]
    member __.DefaultAction() = __.Answer "default"

File: StrategyBot.Game.Logic/Controllers/MainMenuController.cs
using StrategyBot.Game.Core.Controllers;
using StrategyBot.Game.Core.Controllers.Abstractions;
using StrategyBot.Game.Core.Controllers.Answers;
using StrategyBot.Game.Core.Controllers.Handlers;
using StrategyBot.Game.Logic;

namespace StrategyBot.Game.Controllers
{
    [Controller(isMainController: true)]
    public class MainMenuController : ControllerBase
    {
        [Action("screens.main_menu.attack")]
        public IControllerAnswer Test()
        {
            return Answer("test action");
        }

        [Default]
        public IControllerAnswer Default(PlayerState player)
        {
            return Answer("hi, " + player.SocialId);
        }
    }
}